src/lang/checkers/SanityChecker.ml

```diff
@@ -69,12 +77,104 @@ module ScillaSanityChecker
+                 * this point, register this as the first one. *)
+                | [] -> [[loc]]
+                (* Otherwise add this accept statement to the list of accepts
+                 * already seen on each code path reaching this point. *)
+                | _ -> List.map (fun accepts -> loc :: accepts) seen2)
+            | MatchStmt (_ident, branches) ->
+               (* For each branch in the match statement we have a
+                    new code path to "multiply" with the code paths
+                    which already reached this point, so walk each
+                    branch and build all the results into a new list.
+                *)
+               List.fold_left
+                 (fun seen3 (_pattern, branchstmts) ->
+                   match walk seen2 branchstmts with
+                   | [] -> seen3
+                   | seen4 -> seen3 @ seen4)
+                 [] branches
+            | _ -> seen2
+          ) (seen : loc list list) (stmts : stmt_annot list)
+      in
+        walk [] stmts
+    in
+
+    let check_accepts (contr : contract) =
+      let check_transition_accepts (transition : transition) =
+        let transition_accept_groups = List.map List.rev (find_accept_groups transition.tbody) in
+
+        let dup_accept_warning (group : loc list) : unit =
+          (warn1
+             (Core.sprintf "transition %s had a potential code path with duplicate accept statements:\n"
+                (get_id transition.tname) ^
+                String.concat ""
+                  (List.map
+                     (fun loc -> Core.sprintf "  Accept at %s\n" (get_loc_str loc))
+                     group))
+             warning_level_duplicate_accepts
+             (List.hd group))
+        in
+
+        List.iter
+          (fun group ->
+            match List.length group with
+            | 0 -> ()
+            | 1 -> ()
+            | _ -> dup_accept_warning group)
+          transition_accept_groups;
+
+        transition_accept_groups
+      in
+
+      let all_accept_groups =
+        (List.fold_left
+           (fun acc t -> acc @ check_transition_accepts t)
+           [] contr.ctrans)
+      in
+
+      (if List.length all_accept_groups == 0 then
+         (warn0
+            (Core.sprintf "Contract %s had no transitions with accept statement\n"
+               (get_id contr.cname))
+            warning_level_missing_accept));
+
+      (* only return warnings for now, not errors *)
+      []
+    in
+
     (* No repeating names for params. *)
     let e = check_duplicate_ident ER.get_loc (List.map (fun (i, _) -> i) contr.cparams) in
     (* No repeating field names. *)
     let e = e @ check_duplicate_ident ER.get_loc (List.map (fun (i, _, _) -> i) contr.cfields) in
     (* No repeating transition names. *)
     let e = e @ check_duplicate_ident SR.get_loc (List.map (fun t -> t.tname) contr.ctrans) in
+    (* No repeating accept statements *)
+    let e = e @ check_accepts contr in
```

Comment:
This call should be guarded by the `-cf` command-line option, which enables cashflow analysis. See `src/runners/scilla_checker.ml` for an example of how this is done (you may need to pass an appropriate flag to `SanityCheckers.ml`.

Reply:
Need to make sure I understand correctly: do you mean the check should be skipped if `-cf` is specified, or if it isn't specified, or something else?  I'm not yet aware of the relationship between my check and cashflow analysis.

Thanks for the implementation tip - I assume you are referring to this code in `scilla_checker.ml`:

    let cf_info_opt = if cli.cf_flag then Some (check_cashflow typed_cmod) else None in

Reply:
That is indeed the code snippet.

`-cf` activates cashflow analysis, and I think the acceptance checks fits nicely into a generic "suite of cash-related checks". So the check should be enabled when the `-cf` option is specified, but not when it is not.

Reply:
Ah OK of course, that makes perfect sense - thanks!

Reply:
Done.